mailstore-connector/src/com/axonivy/connector/oauth/UserPasswordProvider.java

```diff
@@ -0,0 +1,34 @@
+package com.axonivy.connector.oauth;
+
+/**
+ * The {@code UserPasswordProvider} interface provides methods to retrieve 
+ * user credentials such as username and password for a given store.
+ * <p>
+ * Implementing classes should define the behavior of how user credentials 
+ * are retrieved based on the store name.
+ * </p>
+ */
+public interface UserPasswordProvider {
```

Comment:
@nvvi if you borrow code from the ivy-core, I would at least have expected to have someone from the ivy-core as reviewer. This will surely lead us into support and follow up issues ... there are no tests, no verifications, and finally there is no architecture alignment.